File: java/src/main/java/gov/doc/ntia/sigmf/ext/global/scos/Action.java
package gov.doc.ntia.sigmf.ext.global.scos;

import com.fasterxml.jackson.annotation.*;

import java.io.Serializable;
import java.util.Date;
import java.util.HashMap;
import java.util.Map;

@JsonInclude(JsonInclude.Include.NON_NULL)
public class Action implements Serializable {

    private static final long serialVersionUID = 1L;

    @JsonProperty(value="name", required = true)
    protected String name;

    @JsonProperty(value="description", required =  false)
    protected  String description;

    @JsonProperty(value="summary", required =false)
    protected String summary;

    protected Map<String, Object> otherFields = new HashMap<>();

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    @JsonAnyGetter
    public Map<String, Object> getOtherFields(){
        return otherFields;
    }

    @JsonAnySetter
    public void add(String key, Object value){
        otherFields.put(key, value);
    }

    public String getSummary() {
        return summary;
    }

    public void setSummary(String summary) {
        this.summary = summary;
    }
}

File: Csharp/sigmf/sigmf/ext/global/scos/Action.cs
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Text;

namespace sigmf
{
    public class Action
    {
        private static readonly long serialVersionUID = 1L;

        [JsonProperty("name")]
        protected string name;

        [JsonProperty("description")]
        protected string description;

        [JsonProperty("type")]
        protected string[] type;

        [JsonProperty("type")]
        protected DateTime endTime;

        [JsonProperty("type")]
        protected DateTime startTime;

        [JsonProperty("type")]
        protected string model;

        protected Dictionary<string, Object> otherfields = new Dictionary<string, Object>();

        public string Name { get => name; set => name = value; }
        public string Description { get => description; set => description = value; }
        public string[] Type { get => type; set => type = value; }
        public DateTime EndTime { get => endTime; set => endTime = value; }
        public DateTime StartTime { get => startTime; set => startTime = value; }
        public string Model { get => model; set => model = value; }

        public Dictionary<string, object> getOtherFields()
        {
            return otherfields;
        }
        public void add(string key, object value)
        {
            otherfields.Add(key, value);
        }
    }
}
